SimPEG/EM/Static/Utils/StaticUtils.py

```diff
@@ -943,51 +1350,84 @@ def getSrc_locs(survey):
-        topo = np.zeros(ZC.shape[0])
-        topoCC = np.zeros(ZC.shape[0])
-        for i in range(ZC.shape[0]):
-            ind = np.argmax(ZC[i, :][ACTIND[i, :]])
-            topo[i] = (
-                ZC[i, :][ACTIND[i, :]].max() + mesh.hz[ACTIND[i, :]][ind]*0.5
-                )
-            topoCC[i] = ZC[i, :][ACTIND[i, :]].max()
+        if mesh.dim == 3:
 
-        return mesh2D, topoCC
-
-    elif mesh.dim == 2:
-
-        mesh1D = Mesh.TensorMesh([mesh.hx], [mesh.x0[0]])
-        yc = mesh.gridCC[:, 1]
-        ACTIND = actind.reshape((mesh.vnC[0], mesh.vnC[1]), order='F')
-        YC = yc.reshape((mesh.vnC[0], mesh.vnC[1]), order='F')
-        topo = np.zeros(YC.shape[0])
-        topoCC = np.zeros(YC.shape[0])
-        for i in range(YC.shape[0]):
-            ind = np.argmax(YC[i, :][ACTIND[i, :]])
-            topo[i] = (
-                YC[i, :][ACTIND[i, :]].max() + mesh.hy[ACTIND[i, :]][ind]*0.5
+            mesh2D = Mesh.TensorMesh([mesh.hx, mesh.hy], mesh.x0[:2])
+            zc = mesh.gridCC[:, 2]
+            ACTIND = actind.reshape(
+                (mesh.vnC[0]*mesh.vnC[1], mesh.vnC[2]),
+                order='F'
+                )
+            ZC = zc.reshape((mesh.vnC[0]*mesh.vnC[1], mesh.vnC[2]), order='F')
+            topoCC = np.zeros(ZC.shape[0])
+
+            for i in range(ZC.shape[0]):
+                ind = np.argmax(ZC[i, :][ACTIND[i, :]])
+                if option == "top":
+                    dz = mesh.hz[ACTIND[i, :]][ind] * 0.45
+                elif option == "center":
+                    dz = 0.
+                else:
+                    raise Exception()
+                topoCC[i] = (
+                    ZC[i, :][ACTIND[i, :]].max() + dz
+                    )
+            return mesh2D, topoCC
+
+        elif mesh.dim == 2:
+
+            mesh1D = Mesh.TensorMesh([mesh.hx], [mesh.x0[0]])
+            yc = mesh.gridCC[:, 1]
+            ACTIND = actind.reshape((mesh.vnC[0], mesh.vnC[1]), order='F')
+            YC = yc.reshape((mesh.vnC[0], mesh.vnC[1]), order='F')
+            topoCC = np.zeros(YC.shape[0])
+            for i in range(YC.shape[0]):
+                ind = np.argmax(YC[i, :][ACTIND[i, :]])
+                if option == "top":
+                    dy = mesh.hy[ACTIND[i, :]][ind] * 0.45
+                elif option == "center":
+                    dy = 0.
+                else:
+                    raise Exception()
+                topoCC[i] = (
+                    YC[i, :][ACTIND[i, :]].max() + dy
+                    )
+            return mesh1D, topoCC
+
+    elif mesh._meshType == "TREE":
+        if mesh.dim == 3:
+            uniqXY = uniqueRows(mesh.gridCC[:, :2])
+            npts = uniqXY[0].shape[0]
+            ZC = mesh.gridCC[:,2]
+            topoCC = np.zeros(npts)
+            if option == "top":
+                # TODO: this assume same hz, need to be modified
+                dz = mesh.hz.min() * 0.45
```

Comment:
where did the 0.45 come from? should it be a variable? (or something the user might want to modify??)

Reply:
do not need this.